src/Features/Core/Portable/InlineHints/AbstractInlineTypeHintsService.cs

```diff
@@ -76,15 +76,53 @@ internal abstract class AbstractInlineTypeHintsService : IInlineTypeHintsService
                     continue;
 
                 finalParts.AddRange(suffix);
+                var taggedText = finalParts.ToTaggedText();
+                var displayString = GetDisplayStringFromParts(taggedText);
+
+                TextChange? textChange = null;
+                if (insertSpan is not null)
+                {
+                    textChange = new TextChange(insertSpan.Value, displayString);
+                }
 
                 result.Add(new InlineHint(
-                    span, finalParts.ToTaggedText(),
+                    span, taggedText, textChange,
                     InlineHintHelpers.GetDescriptionFunction(span.Start, type.GetSymbolKey(cancellationToken: cancellationToken))));
             }
 
             return result.ToImmutable();
         }
 
+        private static string GetDisplayStringFromParts(ImmutableArray<TaggedText> taggedTexts)
```

Comment:
what's happening here?  why do we need this?

Reply:
to string together all the pieces from the tagged text, like what is done to display the hint in the InlineHintTag code. It could also be added to the TypeHint definition and avoid doing that, but I didn't want to make the TypeHint too complicated.

Reply:
i really don't like thsi.  it seems to be back inferring the information at a later point.